crates/net/network/src/swarm.rs

```diff
@@ -199,9 +200,16 @@ where
                         }
                         InboundConnectionError::ExceedsLimit(limit) => {
                             // We currently don't have additional capacity to establish a session
-                            // from an incoming connection, so we drop
-                            // the connection.
-                            trace!(target: "net", %limit, ?remote_addr, "Exceeded incoming connection limit; dropping connection");
+                            // from an incoming connection, so we send a disconnect message with
+                            // TooManyPeers reason.
+                            trace!(target: "net", %limit, ?remote_addr, "Exceeded incoming connection limit; disconnecting");
+
+                            tokio::task::spawn(async move {
+                                let sink = LengthDelimitedCodec::default().framed(stream);
+                                let mut stream = UnauthedP2PStream::new(sink);
+                                let _ =
```

Comment:
I'm not sure about the codec @Rjected wdyt

Reply:
fyi I got inspired by other part in the Reth codebase (in `reth-eth-wire`) where `LengthDelimitedCodec` is used, but renamed to  `PassthroughCodec`